modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/logging/CorrelationLoggingTest.java

```diff
@@ -0,0 +1,386 @@
+import org.wso2.carbon.integration.common.utils.mgt.ServerConfigurationManager;
+import org.wso2.carbon.utils.ServerConstants;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.assertEquals;
+
+public class CorrelationLoggingTest extends APIManagerLifecycleBaseTest {
+    private static final Log log = LogFactory.getLog(CorrelationLoggingTest.class);
+    private String apiId;
+    private String applicationId;
+    private ServerConfigurationManager serverConfigurationManager;
+
+
+    private Map<String, String> header = new HashMap<>();
+    BufferedReader bufferedReader;
+
+    private final String API_CONTEXT = "correlationloggingtest";
+    private final String API_VERSION = "1.0.0";
+    private final String APPLICATION_NAME = "CorrelationTestApp";
+    private final String API_END_POINT_POSTFIX_URL = "xmlapi";
+    private String accessToken;
+    private Boolean httpLog,jdbcLog,synapseLog,methodCallsLog;
+
+    @Factory(dataProvider = "userModeDataProvider")
+    public CorrelationLoggingTest(TestUserMode userMode) throws Exception {
+        this.userMode = userMode;
+        byte[] encodedBytes = Base64.encodeBase64(RESTAPITestConstants.BASIC_AUTH_HEADER.getBytes(StandardCharsets.UTF_8));
+        header.put("Authorization", "Basic " + new String(encodedBytes, StandardCharsets.UTF_8));
+        header.put("Content-Type", "application/json");
+        header.put("Accept","application/json" );
+        String correlationLogFilePath =
+                System.getProperty(ServerConstants.CARBON_HOME) + File.separator + "repository" + File.separator
+                        + "logs" + File.separator + "correlation.log";
+        bufferedReader = new BufferedReader(new FileReader(correlationLogFilePath));
+    }
+
+    @DataProvider
+    public static Object[][] userModeDataProvider() {
+        return new Object[][]{new Object[]{TestUserMode.SUPER_TENANT_ADMIN}, new Object[]{TestUserMode.TENANT_ADMIN}};
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void initialize() throws Exception {
+        super.init();
+        AutomationContext superTenantKeyManagerContext =
+                new AutomationContext(APIMIntegrationConstants.AM_PRODUCT_GROUP_NAME,
+                        APIMIntegrationConstants.AM_KEY_MANAGER_INSTANCE, TestUserMode.SUPER_TENANT_ADMIN);
+        serverConfigurationManager = new ServerConfigurationManager(superTenantKeyManagerContext);
+    }
+
+    @Test(groups = {"wso2.am" }, description = "Testing the default correlation configs using the devops API ")
+    public void testRetrieveDefaultCorrelationLoggingConfigsTest() throws Exception {
+
+        //Retrieve default correlation logs configs from the GET method of the configs resource in devops API
+        HttpResponse loggingResponse =
+                HTTPSClientUtils.doGet(getStoreURLHttps() + "api/am/devops/v0/config/correlation", header);
```

Comment:
Shall we define a constant for `api/am/devops/v0/config/correlation`?

Reply:
Sure